### ISSUE TITLE
[Feature] Partial or full refund when removing a residence

### ISSUE THREAD

**Issue by WoodenAxe** [t00thpick1/Residence#7]
There should be a configurable percentage to credit a player when they remove a residence, defaulting to 0 for current behavior.

**Comment by clucky** [comment]
Agreed. A way to do this since the addition of CostEquation would be to recalculate the cost of the residence according to it's size, then multiply the total by the percentage. 

---

For example, PlayerA has a residence of size 10x20x10 and he wants a refund (configured at 90%). The CostEquation for this particular server is: ((XSize \* ZSize) / 20) + YSize.

((XSize \* ZSize) / 20) + YSize
((10 \* 10) / 20) + 20
(100 / 20) + 20
5 + 20
=25

ANS() \* 0.90
25 \* 0.90
=22.5

In this example, the player would be refunded 22.5 monies.
